### PR TITLE
Fix the config node users count

### PR DESCRIPTION
**Pull request by GogoVega**
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Proposed changes

Bad number of `node.users` for Config Node:

- Deleting a Node inside a Subflow does not update the number of users.
- Replacing a Config Node overwrites the user list to empty.
- Remove duplicate "addUser" (L2426), `addNode` do it later.

Other changes:

- `updateConfigNodeUsers` - for either add or remove and either trigger the event

Related PR: #4757.

## Checklist

- [x] I have read the [contribution guidelines](https://github.com/node-red/node-red/blob/master/CONTRIBUTING.md)
- [ ] For non-bugfix PRs, I have discussed this change on the forum/slack team.
- [x] I have run `npm run test` to verify the unit tests pass
- [ ] I have added suitable unit tests to cover the new/changed functionality

### CODE REVIEW

File: packages/node_modules/@node-red/editor-client/src/js/ui/subflow.js [hunk 1]
```diff
@@ -508,10 +508,8 @@ RED.subflow = (function() {
         var activeSubflow = RED.nodes.subflow(id);
 
         RED.nodes.eachNode(function(n) {
-            if (!keepInstanceNodes && n.type == "subflow:"+id) {
-                removedNodes.push(n);
-            }
-            if (n.z == id) {
+            if (n.z === id || (!keepInstanceNodes && n.type === "subflow:" + id)) {
```
Comment (GogoVega) [inline packages/node_modules/@node-red/editor-client/src/js/ui/subflow.js:511]: Should be handled in `RED.nodes.remove`
```suggestion
            if (!keepInstanceNodes && n.type == "subflow:"+id) {
                removedNodes.push(n);
            }
            if (n.z == id) {
```

File: packages/node_modules/@node-red/editor-client/src/js/nodes.js [hunk 2]
```diff
@@ -2648,19 +2651,44 @@ RED.nodes = (function() {
...
+     * their 'users' list is correct.
+     *
+     * Options:
+     * - `action` - Add or remove the node from the Config Node users list. Default `add`.
+     * - `emitEvent` - Emit the `nodes:changes` event. Default true.
+     *
+     * @param {object} node The node in which to check if it contains references
+     * @param {{ action?: "add" | "remove"; emitEvent?: boolean; }} options Options to apply.
+     */
+    function updateConfigNodeUsers(node, options = {}) {
```
Comment (GogoVega) [inline packages/node_modules/@node-red/editor-client/src/js/nodes.js:2665]: Improves the docs
```suggestion
    /**
     * Update any config nodes referenced by the provided node to ensure
     * their 'users' list is correct.
     *
     * @param {object} node The node in which to check if it contains references
     * @param {object} options Options to apply.
     * @param {"add" | "remove"} [options.action] Add or remove the node from
     * the Config Node users list. Default `add`.
     * @param {boolean} [options.emitEvent] Emit the `nodes:changes` event.
     * Default true.
     */
    function updateConfigNodeUsers(node, options) {
```

File: packages/node_modules/@node-red/editor-client/src/js/ui/subflow.js (hunk 1 above)
Comment (knolleary) [reply packages/node_modules/@node-red/editor-client/src/js/ui/subflow.js:511]: Can you clarify if you are proposing a change to your own PR for me to review? Or something else?

Comment (GogoVega) [reply packages/node_modules/@node-red/editor-client/src/js/ui/subflow.js:511]: Exactly, I made a change in this file but after looking at other elements I come to the conclusion that this change should not take place here. So you can commit this revert.

**Pull request merged**